neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -0,0 +1,422 @@
+using Neo.Cryptography.ECC;
+using Neo.IO;
+using Neo.Ledger;
+using Neo.Persistence;
+using Neo.VM;
+using Neo.VM.Types;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Numerics;
+using VMArray = Neo.VM.Types.Array;
+
+namespace Neo.SmartContract.Native.Tokens
+{
+    public class NeoToken : NativeContractBase
+    {
+        public const string ServiceName = "Neo.Native.Tokens.NEO";
+        public static readonly byte[] Script = CreateNativeScript(ServiceName);
+        public static readonly UInt160 ScriptHash = Script.ToScriptHash();
+        public static readonly string[] SupportedStandards = { "NEP-5", "NEP-10" };
+        public const string Name = "NEO";
+        public const string Symbol = "neo";
+        public const int Decimals = 0;
+        public static readonly BigInteger DecimalsFactor = BigInteger.Pow(10, Decimals);
+        public static readonly BigInteger TotalAmount = 100000000 * DecimalsFactor;
+
+        private const byte Prefix_Initialized = 11;
+        private const byte Prefix_Account = 20;
+        private const byte Prefix_Validator = 33;
+        private const byte Prefix_ValidatorsCount = 15;
+
+        internal static bool Main(ApplicationEngine engine)
+        {
+            if (!new UInt160(engine.CurrentContext.ScriptHash).Equals(ScriptHash))
+                return false;
```

Comment:
What possibility does this test elliminate?

Reply:
This avoid to call the syscall directly, without the new "APPCall"

Reply:
So, this test should be moved to `NativeContractBase`, so that its "Main" invoke this real Main (or perhaps Unified Entry Main) safely, without this test. This is more related to NativeContract workings than Token workings. Do you agree?